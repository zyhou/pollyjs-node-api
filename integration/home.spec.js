const path = require("path");

const { Polly } = require("@pollyjs/core");
const { setupPolly } = require("setup-polly-jest");
const NodeHttpAdapter = require("@pollyjs/adapter-node-http");
const FSPersister = require("@pollyjs/persister-fs");

Polly.register(NodeHttpAdapter);
Polly.register(FSPersister);

const { createRequest } = require("./createRequest");

jest.unmock("axios");

describe("Integration: home", () => {
  setupPolly({
    mode: process.env.RECORD ? "record" : "replay",
    adapters: ["node-http"],
    persister: "fs",
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, "./__recordings__")
      }
    }
  });

  it("should not change", async () => {
    const { status, body } = await createRequest().get("/");
    expect(status).toBe(200);
    expect(body).toMatchSnapshot();
  });
});
