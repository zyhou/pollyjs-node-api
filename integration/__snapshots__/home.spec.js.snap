// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: home should not change 1`] = `
Object {
  "photos": Array [
    Object {
      "albumId": 1,
      "id": 1,
      "thumbnailUrl": "https://via.placeholder.com/150/92c952",
      "title": "accusamus beatae ad facilis cum similique qui sunt",
      "url": "https://via.placeholder.com/600/92c952",
    },
    Object {
      "albumId": 1,
      "id": 2,
      "thumbnailUrl": "https://via.placeholder.com/150/771796",
      "title": "reprehenderit est deserunt velit ipsam",
      "url": "https://via.placeholder.com/600/771796",
    },
    Object {
      "albumId": 1,
      "id": 3,
      "thumbnailUrl": "https://via.placeholder.com/150/24f355",
      "title": "officia porro iure quia iusto qui ipsa ut modi",
      "url": "https://via.placeholder.com/600/24f355",
    },
    Object {
      "albumId": 1,
      "id": 4,
      "thumbnailUrl": "https://via.placeholder.com/150/d32776",
      "title": "culpa odio esse rerum omnis laboriosam voluptate repudiandae",
      "url": "https://via.placeholder.com/600/d32776",
    },
    Object {
      "albumId": 1,
      "id": 5,
      "thumbnailUrl": "https://via.placeholder.com/150/f66b97",
      "title": "natus nisi omnis corporis facere molestiae rerum in",
      "url": "https://via.placeholder.com/600/f66b97",
    },
  ],
  "posts": Array [
    Object {
      "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
      "id": 1,
      "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
      "userId": 1,
    },
    Object {
      "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
      "id": 2,
      "title": "qui est esse",
      "userId": 1,
    },
    Object {
      "body": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
      "id": 3,
      "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
      "userId": 1,
    },
    Object {
      "body": "ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit",
      "id": 4,
      "title": "eum et est occaecati",
      "userId": 1,
    },
    Object {
      "body": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
      "id": 5,
      "title": "nesciunt quas odio",
      "userId": 1,
    },
  ],
}
`;
